packages/web3-eth/src/utils/get_revert_reason.ts

```diff
@@ -25,6 +25,45 @@ import { DataFormat, DEFAULT_RETURN_FORMAT } from 'web3-utils';
 import { call } from '../rpc_method_wrappers';
 import { RevertReason, RevertReasonWithCustomError } from '../types';
 
+export const parseTransactionError = (error: unknown, contractAbi?: ContractAbi) => {
```

Comment:
Moved to it's own function so it could make `getRevertReason` more concise and because the new util method `getTransactionError` [uses it](https://github.com/web3/web3.js/blob/36c9b0d950f1b882dcbb28839bda5cd83a9ae5bb/packages/web3-eth/src/utils/get_transaction_error.ts#L39)